### PR TITLE
Refactor sql archive user

### PR DESCRIPTION
**Pull request by henrikstranneheim**
## Description

Refactor sql archive user function.

### Added

- New unarchiving CLI command 
- Tests

### Changed

- Removed un-archiving option from CLI command archive-user
- Moved and renamed archive user function to UpdateHandler

### How to prepare for test
- [ ] ssh to hasta (depending on type of change)
- [ ] activate stage: `us`
- [ ] request trailblazer-stage on hasta: `paxa`
- [ ] install on stage:
    ```Shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_trailblazer -t trailblazer -b refactor_sql_archive_user -a
    ```
- [ ] ssh to clinical-db (depending on type of change)
- [ ] install on stage:
`bash /home/proj/production/servers/resources/clinical-db.scilifelab.se/update-trailblazer-ui-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [x] See below

### Expected test outcome
- [x] See below
- [x] Take a screenshot and attach or copy/paste the output.

## Review
- [x] tests executed by HS
- [x] "Merge and deploy" approved by @ChrOertlin 
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [x] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: tests/cli/test_cli_core.py [hunk 1]
```diff
@@ -218,6 +220,80 @@ def test_users(
...
+    # GIVEN populated Trailblazer database
+
+    # WHEN archiving user
+    cli_runner.invoke(archive_user, [archived_user_email], obj=trailblazer_context)
+
+    # THEN log shows users is already archived
+    assert "already archived" in caplog.text
+
+
+def test_archive_user_when_non_existinng_user(
```
Comment (ChrOertlin) [inline tests/cli/test_cli_core.py:251]:
```suggestion
def test_archive_user_when_non_existing_user(
```

File: tests/conftest.py [hunk 2]
```diff
@@ -72,18 +84,28 @@ def fixture_info_store(store: MockStore) -> Generator[MockStore, None, None]:
 
 @pytest.fixture(name="user_store")
```
Comment (ChrOertlin) [inline tests/conftest.py:85]: should there be a scope here?

File: tests/conftest.py [hunk 3]
```diff
@@ -72,18 +84,28 @@ def fixture_info_store(store: MockStore) -> Generator[MockStore, None, None]:
...
 ) -> Generator[MockStore, None, None]:
     """A Trailblazer database wih a populated user table."""
     StoreHelpers.add_user(email=user_email, name=username, store=store)
+    StoreHelpers.add_user(
+        email=archived_user_email, name=archived_username, is_archived=True, store=store
+    )
     yield store
 
 
 @pytest.fixture(name="sample_store")
```
Comment (ChrOertlin) [inline tests/conftest.py:101]: Scope here as well?

File: trailblazer/cli/core.py [hunk 4]
```diff
@@ -122,31 +123,39 @@ def get_users_from_db(context, name: str, email: str, exclude_archived: bool) ->
...
-    """Archive an existing user identified by it's email."""
-    existing_user = context.obj["trailblazer"].user(email, include_archived=True)
+def archive_user(context, email: str) -> None:
+    """Archive an existing user identified by email."""
+    trailblazer_db: Store = context.obj["trailblazer"]
+    existing_user: user = trailblazer_db.user(email=email, include_archived=True)
 
-    if not existing_user:
-        LOG.error(f"User with email {email} not found")
+    if not is_existing_user(user=existing_user, email=email):
```
Comment (ChrOertlin) [inline trailblazer/cli/core.py:132]:
```suggestion
    if not is_existing_user(user=existing_user, email=email):
        LOG.error(f"User {existing_user} not found")
    
    
```

File: trailblazer/cli/core.py [hunk 5]
```diff
@@ -122,31 +123,39 @@ def get_users_from_db(context, name: str, email: str, exclude_archived: bool) ->
...
+    trailblazer_db: Store = context.obj["trailblazer"]
+    existing_user: user = trailblazer_db.user(email=email, include_archived=True)
 
-    if un_archive:
-        context.obj["trailblazer"].archive_user(existing_user, archive=False)
-        LOG.info(f"User un-archived: {email}")
+    if not is_existing_user(user=existing_user, email=email):
+        return
+    if not existing_user.is_archived:
+        LOG.error(f"User with email {email} not archived")
```
Comment (ChrOertlin) [inline trailblazer/cli/core.py:154]: There lines are repeated from functions above - maybe they can become functions themselves? - DRY

File: trailblazer/cli/utils/user_helper.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+LOG = logging.getLogger(__name__)
+
+
+def is_existing_user(user: User, email: str) -> bool:
+    """Check if supplied user exist in db."""
+    if not user:
+        LOG.error(f"User with email {email} not found")
+        return False
+    return True
```
Comment (ChrOertlin) [inline trailblazer/cli/utils/user_helper.py:13]: ah, I now understand why there was no LOG when checking user. Maybe we can add the other check:
```Python 
is_archived
```
here as well

**Review by ChrOertlin — APPROVED** [review]
Looks good! Some minor suggestions

File: tests/conftest.py (hunk 3 above)
Comment (henrikstranneheim) [reply tests/conftest.py:101]: Fixed

File: trailblazer/cli/utils/user_helper.py (hunk 6 above)
Comment (henrikstranneheim) [reply trailblazer/cli/utils/user_helper.py:13]: Fixed

File: trailblazer/cli/core.py (hunk 5 above)
Comment (henrikstranneheim) [reply trailblazer/cli/core.py]: Fixed

File: tests/conftest.py (hunk 2 above)
Comment (henrikstranneheim) [reply tests/conftest.py]: Done

**Comment by henrikstranneheim** [comment]
### CLI ✅ 
```
⌂44.31 θ81° [henrik.stranneheim@hasta:~] [S_base] 4m20s $  trailblazer unarchive-user jimmy.thelander@scilifelab.se
User with email jimmy.thelander@scilifelab.se has archive status: True
User unarchived: jimmy.thelander@scilifelab.se
⌂40.53 θ82° [henrik.stranneheim@hasta:~] [S_base] 19s $  trailblazer unarchive-user jimmy.thelander@scilifelab.se
User with email jimmy.thelander@scilifelab.se has archive status: False
⌂44.18 θ85° [henrik.stranneheim@hasta:~] [S_base] 22s $  trailblazer archive-user jimmy.thelander@scilifelab.se
User with email jimmy.thelander@scilifelab.se has archive status: False
User archived: jimmy.thelander@scilifelab.se
⌂41.40 θ88° [henrik.stranneheim@hasta:~] [S_base] 20s $  trailblazer archive-user jimmy.thelander@scilifelab.se
User with email jimmy.thelander@scilifelab.se has archive status: True
⌂41.12 θ85° [henrik.stranneheim@hasta:~] [S_base] 20s $  trailblazer archive-user jimmy.the@scilifelab.se
User with email jimmy.the@scilifelab.se not found
```

**Comment by henrikstranneheim** [comment]
### API ✅ 
Screenshot 2023-06-26 at 14 25 00

**Pull request merged**